pykechain/models/input_checks.py

```diff
@@ -173,6 +180,36 @@ def check_datetime(dt: Optional[Union[datetime, str]], key: str) -> Optional[str
     return dt
 
 
+def check_date(dt: Optional[Union[datetime, str]], key: str) -> Optional[str]:
+    """Validate a date value to be a date."""
+    if dt is not None and dt is not empty:
+        if isinstance(dt, str):
+            dt = parse_date(dt)
+        if isinstance(dt, date):
+            dt = dt.strftime("%Y-%m-%d")
+        else:
+            raise IllegalArgumentError(
+                f"`{key}` should be a correctly formatted string or a datetime.date() object, "
+                f'"{dt}" ({type(dt)}) is not.'
+            )
+    return dt
+
+
+def check_time(dt: Optional[Union[datetime, str]], key: str) -> Optional[str]:
+    """Validate a time value to be a time."""
+    if dt is not None and dt is not empty:
+        if isinstance(dt, str):
+            dt = parse_time(dt)
+        if isinstance(dt, time):
+            dt = dt.strftime("%H:%M:%S")
+        else:
+            raise IllegalArgumentError(
+                f"`{key}` should be a correctly formatted string or a datetime.time() object, "
+                f'"{dt}" ({type(dt)}) is not.'
+            )
+    return dt
```

Comment:
In this case i'd consider early return and avoiding negated conditions (it tends to be harder to understand for person reading the code)

Nothing wrong with the code itself. Just something to consider.

The same comment applies to `check_date` function above

```suggestion
    if dt is None or dt is empty:
        return dt
    elif isinstance(dt, str):
        dt = parse_time(dt)
    elif isinstance(dt, time):
        dt = dt.strftime("%H:%M:%S")
    else:
        raise IllegalArgumentError(
            f"`{key}` should be a correctly formatted string or a datetime.time() object, "
            f'"{dt}" ({type(dt)}) is not.'
        )
    return dt
```

Reply:
I think it's a good suggestion, but the `if isinstance(dt, time)` is actually dependent on the previous if to succeed. So first we parse the str in a date form and if that is possible, we then format the string in the expected shape. Otherwise we raise the error. So those two ifs cannot become elifs. So keeping that in mind I have implemented your suggestion.